src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxService.cs

```diff
@@ -417,5 +432,38 @@ internal void PerformOperation(DacFxOperation operation, TaskExecutionMode taskE
         {
             operation.Execute(taskExecutionMode);
         }
+
+        /// <summary>
+        /// Changes telemetry status 
+        /// </summary>
+        private void UpdateTelemetryStatus(bool telemetryEnabled)
+        {
+            try
+            {
+                if (telemetryEnabled)
+                {
+                    DacServices.EnableTelemetry(telemetryApplicationName, serviceVersion);
+                }
+                else
+                {
+                    DacServices.DisableTelemetry();
+                }
+            
+            }
+            catch (Exception ex)
+            {
+                Logger.Write(TraceEventType.Warning, $"Failed to update DacFx telemetry status. telemetry enable: {telemetryEnabled}, error: {ex.Message}");
```

Comment:
FYI - I added extension methods to make this easier (e.g. `Logger.Warning("Message"...`)